synapse/handlers/initial_sync.py

```diff
@@ -318,29 +318,34 @@ async def room_initial_sync(
         )
         is_peeking = member_event_id is None
 
-        user_id = requester.user.to_string()
-
         if membership == Membership.JOIN:
             result = await self._room_initial_sync_joined(
-                user_id, room_id, pagin_config, membership, is_peeking
+                requester, room_id, pagin_config, membership, is_peeking
             )
         elif membership == Membership.LEAVE:
             # The member_event_id will always be available if membership is set
             # to leave.
             assert member_event_id
 
             result = await self._room_initial_sync_parted(
-                user_id, room_id, pagin_config, membership, member_event_id, is_peeking
+                requester,
+                room_id,
+                pagin_config,
+                membership,
+                member_event_id,
+                is_peeking,
             )
 
         account_data_events = []
-        tags = await self.store.get_tags_for_room(user_id, room_id)
+        tags = await self.store.get_tags_for_room(requester.user.to_string(), room_id)
```

Comment:
Keeping the separate `user_id` variable seems fine?

Reply:
I guess, will do